src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,325 @@
+        int pastTaskStatus = Integer.parseInt (content.substring (INDEX_PAST_RECORD_STATUS, INDEX_PAST_RECORD_STATUS + 1));
+        if (content.startsWith ("T")) { //pastRecord has a todo task and add the todo to the task list
+            String todoDescription = "todo " + content.substring (INDEX_PAST_RECORD_STATUS + 4, content.length ());
+            listSize = setTodo (todoDescription, listSize, list, tasks);
+            if (pastTaskStatus == 1) { //if pastTaskStatus is 1, mark it done
+                tasks[listSize - 1].makeAsDone ();
+            }
+        } else if (content.startsWith ("E")) { //pastRecord has a event task and add the event to the task list
+            String eventDescription = "event " + content.substring (INDEX_PAST_RECORD_STATUS + 4, content.length () - 1);
+            eventDescription = eventDescription.replace ("(", "/");
+            listSize = setEvent (eventDescription, listSize, list, tasks);
+            if (pastTaskStatus == 1) { //if pastTaskStatus is 1, mark it done
+                tasks[listSize - 1].makeAsDone ();
+            }
+        } else if (content.startsWith ("D")) { //pastRecord has a deadline task and add the deadline to the task list
+            String deadlineDescription = "event " + content.substring (INDEX_PAST_RECORD_STATUS + 4, content.length () - 1);
+            deadlineDescription = deadlineDescription.replace ("(", "/");
+            listSize = setEvent (deadlineDescription, listSize, list, tasks);
+            if (pastTaskStatus == 1) { //if pastTaskStatus is 1, mark it done
+                tasks[listSize - 1].makeAsDone ();
+            }
+        }
+        content = "";
+        return listSize;
+    }
+
+    private static void printDoneLoadingMessage () {
+        System.out.println ("Done loading past record onto your task list!\n" + "Updating task status based on past record\n" + "updating...\n" + ".\n" + ".\n" + "Done updating! Please type \"List\" to check!");
+        printLine ();
+    }
+
+    private static void printImportingMessage () {
+        System.out.println ("Importing data from previous record" + System.lineSeparator () + "." + System.lineSeparator () + "." + System.lineSeparator () + "." + System.lineSeparator () + "." + System.lineSeparator () + "please wait :)");
+        printLine ();
+    }
+
+    private static void exportTasksAsOutputText (ArrayList<String> list, Task[] tasks) throws IOException {
+        //need to change the filePath as the user's filePath
+        String filePath = "D:\\NUS\\Year 2 Sem 1\\CS2113T\\iP\\iPoutput.txt";
+        File file = new File (filePath);
+        FileWriter fileWriter = new FileWriter (file.getAbsoluteFile ());
+        BufferedWriter bufferedWriter = new BufferedWriter (fileWriter);
+        if (list.size () == 0) {
+            bufferedWriter.write ("");
+        } else {
+            for (int i = 0; i < list.size (); i++) {
+                if (tasks[i].getStatusIcon () == "\u2713") {
```

Comment:
magic string? please refer to:https://nus-cs2113-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html#guideline-name-well